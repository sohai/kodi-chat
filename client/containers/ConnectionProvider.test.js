/* global shallow describe it expect */
/* eslint-disable import/first */
import React from 'react';
import createStore from '../store';
import {
  ConnectionProviderRender,
  mapStateToProps,
  mapDispatchToProps
} from './ConnectionProvider';

const store = createStore();

describe('(Container) ConnectionProvider', () => {
  const state = store.getState();
  const initialProps = mapStateToProps(state);
  const initialActions = mapDispatchToProps(x => x);
  const props = {
    ...initialActions,
    ...initialProps
  };

  it('will receive right props', () => {
    expect(initialProps).toMatchSnapshot();
  });

  it('will receive right actions', () => {
    expect(initialActions).toMatchSnapshot();
  });

  it('should render', () => {
    const wrapper = shallow(<ConnectionProviderRender {...props} />, {
      context: { store }
    });
    expect(wrapper).toMatchSnapshot();
  });
});
