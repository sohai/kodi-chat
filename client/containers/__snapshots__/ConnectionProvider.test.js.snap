// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Container) ConnectionProvider should render 1`] = `""`;

exports[`(Container) ConnectionProvider will receive right actions 1`] = `
Object {
  "error": [Function],
  "start": [Function],
  "startAndSubscribe": [Function],
  "stop": [Function],
}
`;

exports[`(Container) ConnectionProvider will receive right props 1`] = `
Object {
  "available": false,
  "error": null,
  "pending": true,
}
`;
